azurerm/resource_arm_automation_dsc_configuration.go

```diff
@@ -0,0 +1,191 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationDscConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationDscConfigurationCreateUpdate,
+		Read:   resourceArmAutomationDscConfigurationRead,
+		Update: resourceArmAutomationDscConfigurationCreateUpdate,
+		Delete: resourceArmAutomationDscConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[a-zA-Z0-9_]{1,64}$`),
+					`The name length must be from 1 to 64 characters. The name can only contain letters, numbers and underscores.`,
+				),
+			},
+
+			"automation_account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"content_embedded": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"location": locationSchema(),
+
+			"log_verbose": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAutomationDscConfigurationCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationDscConfigurationClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Dsc Configuration creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	accName := d.Get("automation_account_name").(string)
+	contentEmbedded := d.Get("content_embedded").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	logVerbose := d.Get("log_verbose").(bool)
+	description := d.Get("description").(string)
+
+	parameters := automation.DscConfigurationCreateOrUpdateParameters{
+		DscConfigurationCreateOrUpdateProperties: &automation.DscConfigurationCreateOrUpdateProperties{
+			LogVerbose:  utils.Bool(logVerbose),
+			Description: &description,
+			Source: &automation.ContentSource{
+				Type:  automation.EmbeddedContent,
+				Value: &contentEmbedded,
+			},
+		},
+		Location: &location,
+		Name:     &name,
```

Comment:
It will fail [validation](https://github.com/Azure/azure-sdk-for-go/blob/master/services/automation/mgmt/2015-10-31/automation/dscnodeconfiguration.go#L62).